tests/console/validate_zfcp.pm

```diff
@@ -0,0 +1,199 @@
+            my $iscsi_output          = script_output("lsscsi | grep '$device_node_name'");
+            unless ($iscsi_output =~ /^\[(?<hctl>.*)\]\s*$peripheral_type\s+$vendor_model_revision/)
+            {
+                die "SCSI device $device_node_name was not found with peripheral type '$peripheral_type' and " .
+                  "vendor/model/revision '$vendor_model_revision'";
+            }
+
+            # Save H:C:T:L
+            my $hctl = $+{hctl};
+
+            # Validate scsi devices with specific command for zfcp: lszfcp
+            my $fcp_channel = $fcp_device->{fcp_channel};
+            my $wwpn        = $lun->{wwpn};
+            my $bus_wwpn    = "$fcp_channel/$wwpn";
+            validate_script_output("lszfcp -D -b $fcp_channel | grep '$bus_wwpn'", qr/$hctl/);
+
+            # Validate scsi devices with specific command for zfcp: lszdev
+            my $names = $lun->{names};
+            $bus_wwpn = "$fcp_channel:$wwpn";
+            validate_script_output("lszdev --no-headings zfcp-lun | grep '$names'",
+                qr/$bus_wwpn/);
+
+            # Set SCSI devices offline and set back online (other states are also possible)
+            my $state_file = "\"/sys/bus/scsi/devices/$hctl/state\"";
+            validate_script_output("cat $state_file", qr/running/);
+            assert_script_run("echo offline > $state_file",
+                fail_message => "state offline could not be set for SCSI device $hctl");
+            validate_script_output("cat $state_file", qr/offline/);
+            assert_script_run("echo running > $state_file",
+                fail_message => "state online could not be set for SCSI device $hctl");
+            validate_script_output("cat $state_file", qr/running/);
+        }
+    }
+}
+
+# The zfcp_ping and zfcp_show commands can probe ports and retrieve information about ports
+# in the attached storage servers and in interconnect elements such as switches, bridges, and hubs.
+# Because the commands are processed by the SAN management server, information can be obtained
+# about ports and interconnect elements that are not connected to your FCP channel.
+# Thus, zfcp_ping and zfcp_show can help to identify configuration problems in a SAN.
+# The zfcp_show command retrieves information about the SAN topology and details about the SAN components.
+sub investigate_san_fabric {
+    my ($fcp_devices) = shift;
+
+    zypper_call("in libzfcphbaapi0");
+    for my $fcp_device (@{$fcp_devices}) {
+        for my $lun (@{$fcp_devices->{fcp_luns}}) {
+            my $wwpn = $lun->{wwpn};
+            script_run("zfcp_ping -t97 $wwpn >> /tmp/zfcp_ping_wwpns.log");
+        }
+    }
+    upload_logs('/tmp/zfcp_ping_wwpns.log');
+    script_run('zfcp_show -v > /tmp/zfcp_show.log');
+    upload_logs('/tmp/zfcp_show.log');
+}
+
+sub run {
+    select_console 'root-console';
+
+    # FCP devices online
+    record_info('FCP devices', 'Verification of FCP devices online');
```

Comment:
this does not needed. Or it could be rephrased. in the VR it is displayed next to the one from the L180. after you run the record_info you jump into `get_fcp_devices_online` which it does not verify anything but get specific test_data. I would rather remove it completely

Reply:
I agree the visualization is not very good or useful, but it is some point in the test that I wanted to mark somehow, so I want to leave it as it is. I could design the test without this logic, I made it a bit smarter so can tell us when the devices are available by failing, maybe I went too far, but just my point of view (trying new things...I guess)

Reply:
it is not only the visualization. The text is also confusing comparing of what the function does after the record_info. in case you want to leave it, why not to move it in the function and display what you want in this point?

Reply:
I used the record_info in front of some block of code, but if it is more clear, I've moved inside as you suggested.